### PR TITLE
fix(shorebird_cli): camel case flavors can confuse path finding for android

### PR DESCRIPTION
**Pull request by AbhijithKonnayil**
added `kebeb-case` in string extensions and used in appbundle path in android command

## Description

Ref : https://github.com/shorebirdtech/shorebird/issues/1830

`shorebird release android`  command when passed with multi dimensional flavors, resulted in error since shorebird cant find the app bundle. It was due to the mismatch in file name path due to different case.  
The  appbundle is created in kebab-case where as shorebird looks for filename in camelCase.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by bryanoltman** [comment]
Thank you! This is going to require tests.

File: packages/shorebird_cli/lib/src/extensions/string.dart [hunk 1]
```diff
@@ -1,4 +1,9 @@
 extension NullOrEmtpy on String? {
   /// Returns `true` if this string is null or empty.
   bool get isNullOrEmpty => this == null || this!.isEmpty;
+
+  String? get toKebabCase {
+    final exp = RegExp('(?<=[a-z])[A-Z]');
+    return this?.replaceAllMapped(exp, (m) => '-${m.group(0)}').toLowerCase();
+  }
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/extensions/string.dart]: The `NullOrEmpty` extension is not the right place for this

**Review by bryanoltman — CHANGES_REQUESTED** [review]
Please add tests for the new functionality.

File: packages/shorebird_cli/lib/src/extensions/string.dart [hunk 2]
```diff
@@ -2,3 +2,13 @@ extension NullOrEmtpy on String? {
   /// Returns `true` if this string is null or empty.
   bool get isNullOrEmpty => this == null || this!.isEmpty;
 }
+
+extension Case on String {
+  /// returns `sentance-case` (kebab case)
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/extensions/string.dart]:
```suggestion
  /// Converts to kebab-case.
```

File: packages/shorebird_cli/test/src/extensions/string_test.dart [hunk 3]
```diff
@@ -7,4 +7,11 @@ void main() {
     expect(''.isNullOrEmpty, true);
     expect('test'.isNullOrEmpty, false);
   });
+
+   test('ToKebabCase', () async {
+    expect('sentanceCase'.toKebabCase, 'sentance-case');
+    expect('SentanceCase'.toKebabCase, 'sentance-case');
+    expect('sentance_case'.toKebabCase, 'sentance-case');
+    expect('sentance-case'.toKebabCase, 'sentance-case');
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/extensions/string_test.dart]:
```suggestion
    expect('sentenceCase'.toKebabCase, 'sentence-case');
    expect('SentenceCase'.toKebabCase, 'sentence-case');
    expect('sentence_case'.toKebabCase, 'sentence-case');
    expect('sentence-case'.toKebabCase, 'sentence-case');
```

File: packages/shorebird_cli/test/src/extensions/string_test.dart [hunk 4]
```diff
@@ -7,4 +7,11 @@ void main() {
     expect(''.isNullOrEmpty, true);
     expect('test'.isNullOrEmpty, false);
   });
+
+   test('ToKebabCase', () async {
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/extensions/string_test.dart]: What does this do for text containing spaces?

Comment (AbhijithKonnayil) [reply packages/shorebird_cli/test/src/extensions/string_test.dart]: it wont make any change as of now, will make changes to convert space to hypen (-).

**Comment by bryanoltman** [comment]
It looks like there are also some code formatting issues that need to be fixed.

Edit: test/src/extensions/string_test.dart, from https://github.com/shorebirdtech/shorebird/actions/runs/8604701717/job/23579393323?pr=1861

**Review by bryanoltman — APPROVED** [review]

**Pull request merged**